### ISSUE TITLE
Authentication for Gradio webapp breaks app

### ISSUE THREAD

**Issue by rclosson** [gradio-app/gradio#779]
### Describe the bug

auth=("username","password") doesn't work

### Reproduction

import gradio as gr

def greet(name):
    return "Hello " + name + "!!"

iface = gr.Interface(fn=greet, inputs="text", outputs="text")
iface.launch(share=True,auth=("benc","P@ssw0rd123"))

### Screenshot

![image](https://user-images.githubusercontent.com/100968394/156840864-65eaab12-7641-409a-94db-2b2c56ef337d.png)
Note the multiple browsers; I've tested across Firefox, Edge and Chrome. 

### System Info

```shell
Gradio 2.8.7, Running in a Jupyter notebook on a Linux Docker (#103-Ubuntu SMP Fri Nov 26 16:13:00 UTC 2021). I've tested in Chrome, Firefox and Edge with no differences.
```

### Severity

blocker

**Comment by abidlabs** [comment]
Yikes can confirm that this is an issue for me in [colab as well](https://colab.research.google.com/drive/1lnJ-Dg2qgjKHvy6wxeiLyGOZTD60Qxhc?usp=sharing). I see the following error in the front end:

```
index.547d75c3.js:10 
        
       Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at tu (index.547d75c3.js:10:4874)
    at Qs (vendor.66a11843.js:1:6822)
    at new lu (index.547d75c3.js:10:6989)
    at Ku (index.547d75c3.js:15:742)
    at Qs (vendor.66a11843.js:1:7023)
    at new Zu (index.547d75c3.js:15:3590)
    at window.launchGradio (index.547d75c3.js:15:15762)
    at index.547d75c3.js:15:16073
/favicon.ico:1 
        
Failed to load resource: the server responded with a status of 404 ()
```

@aliabid94 or @pngwn can one of you guys take a look at this?

**Comment by abidlabs** [comment]
@rclosson in the meantime, please use a previous version of Gradio, e.g it works if you do: `pip install gradio==2.7.5`

**Comment by rclosson** [comment]
The problem with that solution is that my fully-rendered HTML output (in more complex situations than the minimal problem that I presented here) do not get output properly when share=True is on in that earlier version.

**Comment by aliabid94** [comment]
Fixed in 2.8.8. Let us know if you have any other issues with it!

**Comment by rclosson** [comment]
Testing it in my own environment the issue seems to be resolved. Thank you!
